app/liquid_tags/stackexchange_tag.rb

```diff
@@ -1,24 +1,28 @@
 class StackexchangeTag < LiquidTagBase
   PARTIAL = "liquids/stackexchange".freeze
-  # update API version here and in stackexchange_tag_spec when a new version is out
-  API_URL = "https://api.stackexchange.com/2.2/".freeze
+  REGISTRY_REGEXP = %r{https://(?<subdomain>\w*)(?:\.)?(?:stackexchange.com|stackoverflow.com)/(?<post_type>q|a|questions)/(?<id>\d{1,20})(?:/)?(?:[\w\-]*)}
+  ID_REGEXP = /\A(?<id>\d{1,20})\Z/
+  SITE_REGEXP = /(?<subdomain>\b[a-zA-Z]+\b)/
+  REGEXP_OPTIONS = [REGISTRY_REGEXP, ID_REGEXP, SITE_REGEXP].freeze
+  STACKOVERFLOW_REGEXP = %r{https://stackoverflow.com/(q|a|questions)/\d{1,20}(/)?[\w\-]*}
```

Comment:
I think `\d{1,20}(/)?[\w\-]*`has a bit of a possible missing bit.  My read on this is that we would have a match on "https://stackoverflow.com/q/01234567890123456789-hello-world"

The `(/)?[\w\-]*` can replace `(/)?` with `/?` and I think based on my understanding of the ID, you want `(?:/[\w\-]+)?`